src/Containers/OhmmsPETE/tests/test_Array.cpp

```diff
@@ -73,6 +73,12 @@ TEST_CASE("array NestedContainers", "[OhmmsPETE]")
   CHECK(vec_copy(0).back() == 123);
 }
 
+TEST_CASE("Array::data", "[OhmmsPETE]")
```

Comment:
This doesn't seem like quite enough.
Think about what could go wrong when the code gets modified.
Check some edge cases.
Array<T,1>
First and last indicies, 0 and max.
Array where size of the type is different from double.
populate the array and make sure that the dereference is the data at that location (the same error could be easily made in the hand indexing in the test and the data member function.

Basically the same tests can cover the operator() functions.

Reply:
> Check some edge cases.
Array<T,1>
First and last indicies, 0 and max.
Array where size of the type is different from double.

Not understanding why the type needs to be different from double. Just changed double to float.
Also added value checking.

Reply:
I would say its to make sure future work doesn't somehow only work for double or have some hard coded dependence on the sizeof(double)